service/frontend/admin_handler.go

```diff
@@ -2230,6 +2236,65 @@ func (adh *AdminHandler) archetypeNameToID(archetype chasm.Archetype) (chasm.Arc
 	return archetypeID, nil
 }
 
+func (adh *AdminHandler) GetDynamicConfigurations(
+	ctx context.Context,
+	req *adminservice.GetDynamicConfigurationsRequest,
+) (*adminservice.GetDynamicConfigurationsResponse, error) {
+	requestedKeys := req.GetDynamicConfigKeys()
+	dynamicConfig := make(map[string]*dynamicconfigspb.ConstrainedValues, len(requestedKeys))
+
+	for _, key := range requestedKeys {
+		var dcEntry []dynamicconfig.ConstrainedValue
+		k := dynamicconfig.Key(key)
+		dcEntry = adh.dynamicClient.GetValue(k)
+
+		// get global default value
+		if dcEntry == nil {
+			dcEntry = dynamicconfig.GetDefaultValues(k)
```

Comment:
Do we even want to give back the default value? I was thinking we would return the "raw" values without deserializing them. I'm not sure if the default values are always JSON serializable.

I can look into this a bit more but @dnr can speak to it better than I can.